### ISSUE TITLE
模型训练，使用的是哪一方的资源

### ISSUE THREAD

**Issue by Amber-Believe** [secretflow/secretflow#1635]
### Question Category

Others

### Installation Source

Source

### Secretflow Version

1.10.0b1

### Operating System and Distribution

Ubuntu 22.04.3

### Python Version

Python 3.10.15

### Question Details

```shell
问题：目前是一方参与方提供数据bob，另一方为服务方alice，开启生产模式，训练一个简单分类模型，没有使用GPU，在训练过程中，ray status看到只有object_store_memory有变化，CPU没有变化（alice、bob两方都是），感觉很奇怪，一般来讲用于训练的设备，也就是消耗的资源应该是哪一方的呢？（server端吗）
aggregator = SecureAggregator(alice, [bob])
预期结果：
模型训练的时候。使用到的资源是server方的运算资源（服务器），不会使用数据提供方的资源
```

**Comment by zimu-yuxi** [comment]
方便提供下资源使用情况和运行的代码配置吗

**Comment by Amber-Believe** [comment]
Alice方（服务方）：
#ray start --head --node-ip-address="ip" --port="3000" --include-dashboard=False --disable-usage-stats
import secretflow as sf
import spu
from secretflow.ml.boost.homo_boost import SFXgboost
from secretflow.security.aggregation import SPUAggregator, SecureAggregator
# Check the version of your SecretFlow
print('The version of SecretFlow: {}'.format(sf.__version__))

# In case you have a running secretflow runtime already.
sf.shutdown()

cluster_config ={
    'parties': {
        'alice': {
            # replace with alice's real address.
            'address': 'ip:3100',
            'listen_addr': '0.0.0.0:3100'
        },
        'bob': {
            # replace with bob's real address.
            'address': 'ip:3100',
            'listen_addr': '0.0.0.0:3100'
        },
    },
    'self_party': 'alice'
}
sf.init(address='ip:3000', cluster_config=cluster_config)
alice, bob = sf.PYU('alice'), sf.PYU('bob')

def create_dataset_builder(
    batch_size=32,
    train_split=0.8,
    shuffle=True,
    random_seed=1234,
):
    def dataset_builder(x, stage="train"):
      ....
        # Return
        if stage == "train":
            train_step_per_epoch = len(train_loader)

            return train_loader, train_step_per_epoch
        elif stage == "eval":
            eval_step_per_epoch = len(valid_loader)
            return valid_loader, eval_step_per_epoch

    return dataset_builder

data_builder_dict = {
    bob: create_dataset_builder(
        batch_size=32,
        train_split=0.8,
        shuffle=False,
        random_seed=1234,
    ),
}
from secretflow.ml.nn.core.torch import BaseModule
from torch import nn

class ConvRGBNet(BaseModule):
    ...

from secretflow.ml.nn import FLModel
from secretflow.security.aggregation import SecureAggregator
from torch import nn, optim
from torchmetrics import Accuracy, Precision
from secretflow.ml.nn.core.torch import TorchModel, metric_wrapper, optim_wrapper


device_list = [bob]
aggregator = SecureAggregator(alice, [bob])
# prepare model
num_classes = 5
input_shape = (180, 180, 3)

optim_fn = optim_wrapper(optim.Adam, lr=1e-3)
model_def = TorchModel(
    model_fn=ConvRGBNet,
    loss_fn=nn.CrossEntropyLoss,
    optim_fn=optim_fn,
    metrics=[
        metric_wrapper(
            Accuracy, task="multiclass", num_classes=num_classes, average='micro'
        ),
        metric_wrapper(
            Precision, task="multiclass", num_classes=num_classes, average='micro'
        ),
    ],
)

fed_model = FLModel(
    device_list=device_list,
    model=model_def,
    aggregator=aggregator,
    backend="torch",  # backend support ['tensorflow', 'torch']
    strategy="fed_avg_w",
    random_seed=1234,
)
path_to_flower_dataset = './data/flower_photos'
data = {
    bob: path_to_flower_dataset,
}
history = fed_model.fit(
    data,
    None,
    validation_data=data,
    epochs=2,
    batch_size=32,
    aggregate_freq=2,
    sampler_method="batch",
    random_seed=1234,
    dp_spent_step_freq=1,
    dataset_builder=data_builder_dict,
)

Bob方（数据提供方）
#ray start --head --node-ip-address="ip" --port="3000" --include-dashboard=False --disable-usage-stats
import secretflow as sf
import spu
from secretflow.ml.boost.homo_boost import SFXgboost
from secretflow.security.aggregation import SPUAggregator, SecureAggregator
print('The version of SecretFlow: {}'.format(sf.__version__))
sf.shutdown()

cluster_config ={
    'parties': {
        'alice': {
            # replace with alice's real address.
            'address': 'ip:3100',
            'listen_addr': '0.0.0.0:3100'
        },
        'bob': {
            # replace with bob's real address.
            'address': 'ip:3100',
            'listen_addr': '0.0.0.0:3100'
        },
    },
    'self_party': 'bob'
}
sf.init(address='ip:3000', cluster_config=cluster_config,)
alice, bob = sf.PYU('alice'), sf.PYU('bob')

def create_dataset_builder(
    batch_size=32,
    train_split=0.8,
    shuffle=True,
    random_seed=1234,
):
    def dataset_builder(x, stage="train"):
      .....
        # Return
        if stage == "train":
            train_step_per_epoch = len(train_loader)

            return train_loader, train_step_per_epoch
        elif stage == "eval":
            eval_step_per_epoch = len(valid_loader)
            return valid_loader, eval_step_per_epoch

    return dataset_builder
data_builder_dict = {
    bob: create_dataset_builder(
        batch_size=32,
        train_split=0.8,
        shuffle=False,
        random_seed=1234,
    ),
}
from secretflow.ml.nn.core.torch import BaseModule
from torch import nn

class ConvRGBNet(BaseModule):
   ......

from secretflow.ml.nn import FLModel
from secretflow.security.aggregation import SecureAggregator
from torch import nn, optim
from torchmetrics import Accuracy, Precision
from secretflow.ml.nn.core.torch import TorchModel, metric_wrapper, optim_wrapper


device_list = [bob]
aggregator = SecureAggregator(alice, [bob])
# prepare model
num_classes = 5
input_shape = (180, 180, 3)
optim_fn = optim_wrapper(optim.Adam, lr=1e-3)
model_def = TorchModel(
    model_fn=ConvRGBNet,
    loss_fn=nn.CrossEntropyLoss,
    optim_fn=optim_fn,
    metrics=[
        metric_wrapper(
            Accuracy, task="multiclass", num_classes=num_classes, average='micro'
        ),
        metric_wrapper(
            Precision, task="multiclass", num_classes=num_classes, average='micro'
        ),
    ],
)

fed_model = FLModel(
    device_list=device_list,
    model=model_def,
    aggregator=aggregator,
    backend="torch",  # backend support ['tensorflow', 'torch']
    strategy="fed_avg_w",
    random_seed=1234,

)
path_to_flower_dataset = './data/flower_photos'
data = {
    bob: path_to_flower_dataset,
}
history = fed_model.fit(
    data,
    None,
    validation_data=data,
    epochs=2,
    batch_size=32,
    aggregate_freq=2,
    sampler_method="batch",
    random_seed=1234,
    dp_spent_step_freq=1,
    dataset_builder=data_builder_dict,
)


后台情况：
Alice:
![image](https://github.com/user-attachments/assets/2dd48afc-699a-493e-9e2e-9ac7d9a7ec32)
代码运行过程中，也就是训练中，只有object_store_memory有变化
![image](https://github.com/user-attachments/assets/79133595-2ac1-4c04-9e7f-d05c5aade144)

Bob方：
![image](https://github.com/user-attachments/assets/66336b4f-3f28-4abf-9c1c-c107af09940b)

![image](https://github.com/user-attachments/assets/9f8a77cd-9703-46d0-beec-d4273e6ea6bb)

**Comment by zimu-yuxi** [comment]
训练过程中top看下ray使用cpu的情况呢

**Comment by Amber-Believe** [comment]
![image](https://github.com/user-attachments/assets/e3284283-a7df-4113-9dad-92ca34a92cc5)

**Comment by Amber-Believe** [comment]
BOB方
![image](https://github.com/user-attachments/assets/23405239-d7c7-4c14-8b18-046a8f69b88e)

**Comment by zimu-yuxi** [comment]
可以参考下[这里](https://www.secretflow.org.cn/zh-CN/docs/secretflow/v1.10.0b1/tutorial/nn_with_spu)

**Comment by Amber-Believe** [comment]
还是想咨询下，正常使用训练资源是用的那块的（哪一方的，还是双方）？object_store_memory是提供的“多方的虚拟空间”吗？方便解答下吗
